From 0b63e9938261e047604321ca6cde3f85a7d67449 Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Sun, 29 May 2016 20:38:39 -0400
Subject: [PATCH 1573/1722] drm/amdkfd: Remove duplicate pqm_uninit()

pqm_uninit() will be called in kfd_process_notifier_release(), which
is when the process exits. Calling it in kfd_unbind_process_from_device()
is duplicate and in fact incorrect, because pqm should not be
uninitalized as long as the process still exists.

Change-Id: I65f5095c0e1f346bda76d429dd8d88ff5c3eae0c
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_process.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_process.c b/drivers/gpu/drm/amd/amdkfd/kfd_process.c
index 6964174..0f5f78e 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_process.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_process.c
@@ -706,8 +706,6 @@ void kfd_unbind_process_from_device(struct kfd_dev *dev, unsigned int pasid)
 
 	mutex_unlock(get_dbgmgr_mutex());
 
-	pqm_uninit(&p->pqm);
-
 	pdd = kfd_get_process_device_data(dev, p);
 	if (pdd->reset_wavefronts) {
 		dbgdev_wave_reset_wavefronts(pdd->dev, p);
-- 
2.7.4

